### ISSUE TITLE
@types/nightwatch types for BDD testing

### ISSUE THREAD

**Issue by sinziananicolae** [DefinitelyTyped/DefinitelyTyped#51549]
- [x] I tried using the `@types/nightwatch` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @rkavalap @schlesiger @ClaytonAstrom @lloiser 

@types/nightwatch doesn't currently support the types for BDD testing https://nightwatchjs.org/guide/using-nightwatch/using-bdd-describe.html (describe, it, test etc). For my project I also have Jest and the types for them get inferred from Jest which makes it worse. Is there any plan to support this?

**Comment by lloiser** [comment]
I have never used it before, so I don't know exactly how to setup or what types are involved.
But I will definitely do a review if you want to add those type definitions yourself :)

**Comment by vaibhavsingh97** [comment]
@sinziananicolae Thanks for raising the issue 🙌 . I know it's stuck here for a long time, but finally, we had added the types for BDD style testing. Here is how you can write test in typescript

```typescript
import {
  describe,
  it,
  before,
  after,
  xit,
  xdescribe,
  test,
} from 'nightwatch';

describe('Ecosia', () => {
  before((browser) => browser.url('https://www.ecosia.org/'));

  it('Demo test ecosia.org', () => {
    browser
      .waitForElementVisible('body')
      .assert.titleContains('Ecosia')
      .assert.titleContains('Ecosia')
      .assert.visible('input[type=search]')
      .setValue('input[type=search]', 'nightwatch')
      .assert.visible('button[type=submit]')
      .click('button[type=submit]');
  });

  xit('this test will be skipped', () => {
    browser.waitForElementVisible('body');
  });

  after((browser) => browser.end());
});

xdescribe('whole describle block will be skipped', () => {
  test('ecosia', () => {
    browser.url('https://ecosia.org').end();
  });
});

describe('Async Ecosia', () => {
  before((browser) => browser.url('https://www.ecosia.org/'));

  it('Demo test ecosia.org', async () => {
    browser.waitForElementVisible('body');
  });

  after((browser) => browser.end());
});
```

It will be available in [2.0.6](https://npmjs.com/package/@types/nightwatch/v/2.0.6). Please feel to raise the issue again, if you feel that something is broken.  


PS: If you want to discuss about Nightwatch, feel free to join our Discord (https://discord.gg/A4JmdtpE).

**Comment by kkallu1** [comment]
Hi @vaibhavsingh97 

I am facing   **TypeError: (0 , nightwatch_1.describe) is not a function.**  please let me know how to fix it ,

TIA
